client/state/data-layer/http-data/getters.js

```diff
@@ -0,0 +1,38 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import makeJsonSchemaParser from 'lib/make-json-schema-parser';
+import { http as rawHttp } from 'state/http/actions';
+import { requestHttpData } from 'state/data-layer/http-data/common';
+
+export const requestGeoLocation = () =>
+	requestHttpData(
+		'geo',
+		rawHttp( {
+			method: 'GET',
+			url: 'https://public-api.wordpress.com/geo/',
+		} ),
+		{
+			fromApi: makeJsonSchemaParser(
+				{
+					type: 'object',
+					properties: {
+						body: {
+							type: [ 'object', 'null' ],
+							properties: {
+								latitude: { type: 'string' },
+								longitude: { type: 'string' },
+								country_short: { type: 'string' },
+								country_long: { type: 'string' },
+								region: { type: 'string' },
+								city: { type: 'string' },
+							},
+						},
+					},
+				},
+				// we only use the short code currently
+				( { body: { country_short } } ) => [ [ 'geo', country_short ] ]
```

Comment:
The schema says that `body` can be `null` and yet we destructure it here like it could never fail.

Reply:
technically as part of `fromApi` it should failover automatically if we were to get a type error. I will update the schema and remove the parser